.github/workflows/build.yml

```diff
@@ -0,0 +1,176 @@
+name: CI
+
+on:
+  push:
+  pull_request_target:
+
+permissions:
+  contents: read
+  packages: write
+
+env:
+  DOCKER_DRIVER: overlay2
+  FAST_MODE: false
+
+jobs:
+  builder-image:
+    name: Builder Image
+    runs-on: ubuntu-20.04
+    outputs:
+      image-tag: ${{ steps.prepare.outputs.image-tag }}
+      repo-name: ${{ steps.prepare.outputs.repo-name }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Prepare
+        id: prepare
+        run: |
+          BRANCH_NAME=$(echo "${GITHUB_REF##*/}" | tr '[:upper:]' '[:lower:]')
```

Comment:
why do we need in a step of lower-case-iation?

Reply:
yes, because docker images revolt if it's not lowercase